p4-16/spec/P4-16-spec.mdk

```diff
@@ -2207,8 +2207,7 @@ target-specific).
 It is also possible to specify an `enum` with an underlying representation.
 These are sometimes called serializable `enum`s, because headers are
 allowed to have fields with such `enum` types.
-This requires the programmer provide both the fixed-width unsigned (or signed) integer type and
-an associated integer value for each symbolic entry in the enumeration.  For example, the
+This requires the programmer provide both the fixed-width unsigned (or signed) integer type (or their nested arbitrarily `typedef`) and an associated integer value for each symbolic entry in the enumeration.  For example, the
```

Comment:
I think it would be clearer to add something like this here, instead of the changes you suggest:

The symbol `typeRef` in the grammar above must be one of the following types:
+ an unsigned integer, i.e. `bit<W>` for some constant `W`
+ a signed integer, i.e. `int<W>` for some constant `W`
+ a type name declared via `typedef`, where the base type of that type is either one of the types listed above, or another `typedef` name that meets these conditions.